cmd/cli/kubectl-kyverno/utils/report/report.go

```diff
@@ -14,71 +10,25 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const clusterpolicyreport = "clusterpolicyreport"
-
-// resps is the engine responses generated for a single policy
-func buildPolicyReports(auditWarn bool, engineResponses ...engineapi.EngineResponse) ([]policyreportv1alpha2.ClusterPolicyReport, []policyreportv1alpha2.PolicyReport) {
-	var clustered []policyreportv1alpha2.ClusterPolicyReport
-	var namespaced []policyreportv1alpha2.PolicyReport
-	resultsMap := buildPolicyResults(auditWarn, engineResponses...)
-	for scope, result := range resultsMap {
-		if scope == clusterpolicyreport {
-			report := policyreportv1alpha2.ClusterPolicyReport{
-				TypeMeta: metav1.TypeMeta{
-					APIVersion: policyreportv1alpha2.SchemeGroupVersion.String(),
-					Kind:       "ClusterPolicyReport",
-				},
-				Results: result,
-				Summary: reportutils.CalculateSummary(result),
-			}
-			report.SetName(scope)
-			clustered = append(clustered, report)
-		} else {
-			report := policyreportv1alpha2.PolicyReport{
-				TypeMeta: metav1.TypeMeta{
-					APIVersion: policyreportv1alpha2.SchemeGroupVersion.String(),
-					Kind:       "PolicyReport",
-				},
-				Results: result,
-				Summary: reportutils.CalculateSummary(result),
-			}
-			policyNamespace := strings.ReplaceAll(scope, "policyreport-ns-", "")
-			report.SetName(scope)
-			report.SetNamespace(policyNamespace)
-			namespaced = append(namespaced, report)
-		}
-	}
-
-	return clustered, namespaced
-}
-
-// buildPolicyResults returns a string-PolicyReportResult map
-// the key of the map is one of "clusterpolicyreport", "policyreport-ns-<namespace>"
-func buildPolicyResults(auditWarn bool, engineResponses ...engineapi.EngineResponse) map[string][]policyreportv1alpha2.PolicyReportResult {
-	results := make(map[string][]policyreportv1alpha2.PolicyReportResult)
-	now := metav1.Timestamp{Seconds: time.Now().Unix()}
-
+func ComputePolicyReportResultsPerPolicy(auditWarn bool, engineResponses ...engineapi.EngineResponse) map[engineapi.GenericPolicy][]policyreportv1alpha2.PolicyReportResult {
+	results := make(map[engineapi.GenericPolicy][]policyreportv1alpha2.PolicyReportResult)
 	for _, engineResponse := range engineResponses {
+		if len(engineResponse.PolicyResponse.Rules) == 0 {
+			continue
+		}
 		policy := engineResponse.Policy()
 		policyName := policy.GetName()
-		policyNamespace := policy.GetNamespace()
+		audit := engineResponse.GetValidationFailureAction().Audit()
 		scored := annotationsutils.Scored(policy.GetAnnotations())
 		category := annotationsutils.Category(policy.GetAnnotations())
 		severity := annotationsutils.Severity(policy.GetAnnotations())
-
-		var appname string
-		if policyNamespace != "" {
-			appname = fmt.Sprintf("policyreport-ns-%s", policyNamespace)
-		} else {
-			appname = clusterpolicyreport
-		}
-
 		for _, ruleResponse := range engineResponse.PolicyResponse.Rules {
+			// TODO only validation is managed here ?
```

Comment:
We support the validate and imageVerify rules for reporting if running in-cluster. Reports for validate is supported in CLI for sure, not sure about the imageVerify rule.

Reply:
Yeah thanks, this file was moved (just fixed a few bugs) so this should not be changed compared to before.
I added the TODO to not forget double checking when using it in the `test` command.
If something is wrong it will show up.